src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -1098,6 +1103,59 @@ public void closeChannel() {
     }
   }
 
+  /**
+   * Asynchronously closes a channel associated to this partition. Any {@link SFException} occurred
+   * is swallowed and a successful {@link CompletableFuture} is returned instead.
+   */
+  public CompletableFuture<Void> closeChannelAsync() {
+    return closeChannelWrapped()
+        .thenAccept(__ -> onCloseChannelSuccess())
+        .exceptionally(this::tryRecoverFromCloseChannelError);
+  }
+
+  private CompletableFuture<Void> closeChannelWrapped() {
+    try {
+      return this.channel.close();
+    } catch (SFException e) {
+      // Calling channel.close() can throw an SFException instead of putting it into a completable
+      // future.
+      CompletableFuture<Void> future = new CompletableFuture<>();
+      future.completeExceptionally(e);
+      return future;
```

Comment:
A bit shorter
```suggestion
      return CompletableFuture.failedFuture(e);
```

Reply:
`CompletableFuture.failedFuture` was introduced in Java 9 only :(

Reply:
Oh :C